<?xml version="1.0"?>
<launch>
	<arg name="ns" default="robot_0"/>

    <!-- **************************** tf for Gmapping **************************** -->
	<node pkg="tf" type="static_transform_publisher" name="laserscan_tf" args="0.2 0 0.35 0 0 0 /$(arg ns)/base_link /$(arg ns)/base_scan 33" />

	<!-- **************************** gmapping ****************************-->
	<node pkg="gmapping" type="slam_gmapping" name="gmapping">
        <rosparam file="$(find pioneer3at)/config/gmapping/gmapping_stage.yaml" command="load" />
	    <param name="base_frame" type="str" value="/$(arg ns)/base_link" />
	    <param name="map_frame" type="str" value="/$(arg ns)/map" />
	    <param name="odom_frame" type="str" value="/$(arg ns)/odom" />
	    <param name="occ_map_frame" type="str" value="/$(arg ns)/occ_map" />
	    <remap from="scan"          to="/$(arg ns)/base_scan"/>
	</node>

	<!-- **************************** move_base ****************************-->
	<node pkg="move_base" type="move_base" name="move_base" output="screen">
		<rosparam file="$(find pioneer3at)/config/move_base/move_base_multi.yaml" command="load"/>
		<rosparam file="$(find pioneer3at)/config/move_base/costmap_common_multi.yaml" command="load" ns="global_costmap" />
		<rosparam file="$(find pioneer3at)/config/move_base/costmap_common_multi.yaml" command="load" ns="local_costmap" />
		<rosparam file="$(find pioneer3at)/config/move_base/costmap_global_multi.yaml" command="load"  />
		<rosparam file="$(find pioneer3at)/config/move_base/costmap_local_multi.yaml" command="load"/>

	    <param name="move_base/global_frame_id" type="str" value="$(arg ns)/map" />
	    <param name="global_costmap/global_frame" type="str" value="$(arg ns)/map" />
	    <param name="local_costmap/global_frame" type="str" value="$(arg ns)/map" />
	    <param name="global_costmap/map_topic" type="str" value="$(arg ns)/map" />
	    <param name="local_costmap/map_topic" type="str" value="$(arg ns)/map" />
	    <param name="global_costmap/robot_base_frame" type="str" value="$(arg ns)/base_link" />
	    <param name="global_costmap/laserscan/topic" type="str" value="$(arg ns)/base_scan" />
	    <param name="global_costmap/laserscan/sensor_frame" type="str" value="$(arg ns)/base_scan" />
	    <param name="local_costmap/robot_base_frame" type="str" value="$(arg ns)/base_link" />
	    <param name="local_costmap/laserscan/topic" type="str" value="$(arg ns)/base_scan" />
	    <param name="local_costmap/laserscan/sensor_frame" type="str" value="$(arg ns)/base_scan" />
		<param name="controller_frequency" value="10.0" />
	</node>

	<!-- **************************** Explorer ****************************-->
	<node pkg="pioneer3at" type="explorer" name="Explorer" required="true">
	    <param name="robot_topic" type="str" value="/$(arg ns)" />
	    <param name="goal_topic" type="str" value="/$(arg ns)/move_base_simple/goal" />
	    <param name="occ_map_topic" type="str" value="/$(arg ns)/occ_map" />
	    <param name="map_topic" type="str" value="/$(arg ns)/map" />
	    <param name="pose_topic" type="str" value="/$(arg ns)/odom" />
	    <param name="cmd_vel_topic" type="str" value="/$(arg ns)/cmd_vel" />
	    <param name="laser_topic" type="str" value="/$(arg ns)/base_scan" />
	    <param name="base_link_topic" type="str" value="/$(arg ns)/base_link" />
	    <param name="goal_status_topic" type="str" value="/$(arg ns)/move_base/status" />
		<rosparam file="$(find pioneer3at)/config/explorer/test.yaml" command="load" />
	</node>
</launch>

